### ISSUE TITLE
CoreML mlmodel export boxes are too large

### ISSUE THREAD

**Issue by sidekickr** [ultralytics/ultralytics#20356]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Export

### Bug

When using 8.3.116 the `mlmodel` export is producing boxes that are too large. See attached image from the Xcode preview. Export in 8.3.115 works correctly. Debugging shows that removing this [line](https://github.com/ultralytics/ultralytics/blob/dc9b0c1e759ac6f7a6440c5cdd7b476146fb03a3/ultralytics/engine/exporter.py#L387) resolves the issue. See the second image for what it looks like with that line commented out. 

I'm not sure why this line matters for CoreML exports, but it does. It should probably be excluded for all CoreML exports. I can prepare a PR for this if desired.

<img width="631" alt="Image" src="https://github.com/user-attachments/assets/b3ca48aa-6d9d-4826-91cb-78dab8a5f95b" />

<img width="622" alt="Image" src="https://github.com/user-attachments/assets/32f1bd99-f2dc-4ac8-b33d-c9c6c807e7d6" />

### Environment

Ultralytics 8.3.116 🚀 Python-3.11.5 torch-2.3.1 CPU (Apple M1 Max)
Setup complete ✅ (10 CPUs, 64.0 GB RAM, 1608.3/1858.2 GB disk)

OS                  macOS-15.4.1-arm64-arm-64bit
Environment         Darwin
Python              3.11.5
Install             git
Path                /Users/davidsiebecker/Developer/ultralytics/ultralytics
RAM                 64.00 GB
Disk                1608.3/1858.2 GB
CPU                 Apple M1 Max
CPU count           10
GPU                 None
GPU count           None
CUDA                None

numpy               ✅ 1.23.5<=2.1.1,>=1.23.0
matplotlib          ✅ 3.8.4>=3.3.0
opencv-python       ✅ 4.9.0.80>=4.6.0
pillow              ✅ 10.3.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.13.0>=1.4.1
torch               ✅ 2.3.1>=1.8.0
torch               ✅ 2.3.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.18.1>=0.9.0
tqdm                ✅ 4.66.2>=4.64.0
psutil              ✅ 5.9.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.1>=1.1.4
seaborn             ✅ 0.12.2>=0.11.0
ultralytics-thop    ✅ 2.0.0>=2.0.0

### Minimal Reproducible Example

from ultralytics import YOLO, checks, hub, utils

# checks()

model = YOLO('/opt/homebrew/weights/hub/<modelid>/best.pt')
model.export(format="mlmodel",nms=True,half=False)

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sidekickr, thank you for reporting this and for your interest in contributing to Ultralytics 🚀! This is an automated response — an Ultralytics engineer will review your report and assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and answers to common questions.

Since this is a 🐛 Bug Report, please make sure your [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) is complete so we can reproduce the issue as efficiently as possible.

Join the Ultralytics community! For real-time chat, visit [Discord](https://discord.com/invite/ultralytics) 🎧. For more in-depth discussions, check out [Discourse](https://community.ultralytics.com/), or share your experience on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Please upgrade to the latest `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to see if your issue persists:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following up-to-date verified environments, with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your detailed report and willingness to submit a PR! An engineer will follow up with you for any additional details needed.

**Comment by Y-T-G** [comment]
@sidekickr Can you open a PR for it?

You can change it to `m.xyxy = self.args.nms and not coreml`

**Comment by sidekickr** [comment]
@Y-T-G PR is https://github.com/ultralytics/ultralytics/pull/20357

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in [PR #20357](https://github.com/ultralytics/ultralytics/pull/20357), titled "Fix box sizes for CoreML exports," authored by @sidekickr with contributions from @Y-T-G.  

### Key Changes  
This PR improves the export logic, specifically targeting CoreML compatibility:  
- Adjusts the `xyxy` parameter to ensure compatibility by disabling it for CoreML exports, while maintaining the existing logic for other formats like ONNX and TensorFlow.  
- Aims to provide smoother and more reliable exports to CoreML, reducing deployment errors without impacting workflows for other formats.

To test these changes:  
1. Install the latest updates directly from the repository:  
   ```
   pip install git+https://github.com/ultralytics/ultralytics.git@main
   ```  
2. Alternatively, you can wait for the next official release on PyPI to test this fix.  

Please let us know if this resolves the issue. Your feedback is incredibly valuable in ensuring smooth functionality! Thank you for reporting this and helping improve the project. 🙏 If the issue persists, feel free to share more details.
